### PR TITLE
Version 2.2.3

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
pandas 2.2.3

**Destination channel:** defaults

### Links

- [PKG-4449](https://anaconda.atlassian.net/browse/PKG-4449) 
- [Upstream repository](https://github.com/pandas-dev/pandas/tree/v2.2.3)
- [Upstream changelog/diff](https://github.com/pandas-dev/pandas/tree/v2.2.3)
- Relevant dependency PRs:
  - ...

### Explanation of changes:
- ...

[PKG-4449]: https://anaconda.atlassian.net/browse/PKG-4449?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,8 +25,8 @@ requirements:
     - meson-python ==0.13.1
     - meson ==1.2.1
     - cython >=3.0.5,<4
-    - numpy 1.22  # [py<311]
-    - numpy {{ numpy }}  # [py>=311]
+    - numpy >=1.22,<2.0  # [py<310]
+    - numpy >=2.0,<2.3  # [py>=310]
     - versioneer
     - pip
   run:
```
Comment (cbouss) [inline recipe/meta.yaml:32]: Can you adjust the numpy depenedency in the run section?
When built with numpy 2, pin_compatible is too restrictive.

**Comment by cbouss** [comment]
Can you also add this to abs.yaml to enable 3.13 builds?
```
build_env_vars:
  ANACONDA_ROCKET_ENABLE_PY313 : yes
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -25,14 +25,15 @@ requirements:
     - meson-python ==0.13.1
     - meson ==1.2.1
     - cython >=3.0.5,<4
-    - numpy 1.22  # [py<311]
-    - numpy {{ numpy }}  # [py>=311]
+    - numpy >=1.22,<2.0  # [py<310]
+    - numpy >=2.0,<2.3  # [py>=310]
```
Comment (cbouss) [inline recipe/meta.yaml:29]:
```suggestion
    - numpy 2.0 # [py<313]
    - numpy 2.1  # [py>=313]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -25,14 +25,15 @@ requirements:
...
     - versioneer
     - pip
   run:
     - python
     # Base Dependencies
-    - {{ pin_compatible('numpy') }}
+    - numpy >=1.22,<2.0  # [py<310]
+    - numpy >=2.0,<2.3  # [py>=310]
```
Comment (cbouss) [inline recipe/meta.yaml:36]:
```suggestion
    - numpy>=1.22.4,<3 # [py<311]
    - numpy>=1.23.2,<3 # [py==311]
    - numpy>=1.26.0,<3 # [py==312]
    - numpy>=2.1,<3 # [py==313]
```

**Review by cbouss — COMMENTED** [review]
Added suggestions based on pyproject.toml content.

**Review by ViridianMelody — APPROVED** [review]
Agree on the `numpy` pinning suggestions by Charles, otherwise looks good!

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**